src/main/java/org/elasticsearch/search/aggregations/reducers/bucketmetrics/MaxBucketReducer.java

```diff
@@ -0,0 +1,163 @@
+import org.elasticsearch.search.aggregations.support.format.ValueFormatterStreams;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class MaxBucketReducer extends SiblingReducer {
+
+    public final static Type TYPE = new Type("max_bucket");
+
+    public final static ReducerStreams.Stream STREAM = new ReducerStreams.Stream() {
+        @Override
+        public MaxBucketReducer readResult(StreamInput in) throws IOException {
+            MaxBucketReducer result = new MaxBucketReducer();
+            result.readFrom(in);
+            return result;
+        }
+    };
+
+    private ValueFormatter formatter;
+
+    public static void registerStreams() {
+        ReducerStreams.registerStream(STREAM, TYPE.stream());
+    }
+
+    private MaxBucketReducer() {
+    }
+
+    protected MaxBucketReducer(String name, String[] bucketsPaths, @Nullable ValueFormatter formatter, Map<String, Object> metaData) {
+        super(name, bucketsPaths, metaData);
+        this.formatter = formatter;
+    }
+
+    @Override
+    public Type type() {
+        return TYPE;
+    }
+
+    public InternalAggregation doReduce(Aggregations aggregations, ReduceContext context) {
+        List<String> maxBucketKeys = new ArrayList<>();
+        double maxValue = Double.MIN_VALUE;
+        List<String> bucketsPath = AggregationPath.parse(bucketsPaths()[0]).getPathElementsAsStringList();
+        for (Aggregation aggregation : aggregations) {
+            if (aggregation.getName().equals(bucketsPath.get(0))) {
+                bucketsPath = bucketsPath.subList(1, bucketsPath.size());
+                InternalMultiBucketAggregation multiBucketsAgg = (InternalMultiBucketAggregation) aggregation;
+                List<? extends Bucket> buckets = multiBucketsAgg.getBuckets();
+                for (int i = 0; i < buckets.size(); i++) {
+                    Bucket bucket = buckets.get(i);
+                    Double bucketValue = resolveBucketValue(multiBucketsAgg, bucket, bucketsPath);
+                    if (bucketValue != null) {
+                        if (bucketValue > maxValue) {
+                            maxBucketKeys.clear();
+                            maxBucketKeys.add(bucket.getKeyAsString());
+                            maxValue = bucketValue;
+                        } else if (bucketValue == maxValue) {
+                            maxBucketKeys.add(bucket.getKeyAsString());
+                        }
+                    }
+                }
+            }
+        }
+        String[] keys = maxBucketKeys.toArray(new String[maxBucketKeys.size()]);
+        return new InternalBucketMetricValue(name(), keys, maxValue, formatter, Collections.EMPTY_LIST, metaData());
+    }
+
+    private Double resolveBucketValue(InternalMultiBucketAggregation multiBucketAggregation, InternalMultiBucketAggregation.Bucket bucket,
```

Comment:
I actually thought I already had unified this with the method in BucketHelpers but will look. Thanks for catching